### PR TITLE
Go sync

### LINKED ISSUE #12: Add getSync method for simplicity in configs
Because you can!

### PR DESCRIPTION
**Pull request by tadatuta**
Closes #12

### CODE REVIEW

**Comment by tadatuta** [comment]
@blond @zxqfox @Yeti-or, if such implementation is fine for you, I'll add tests.

**Comment by zxqfox** [comment]
Implementation looks fine.

What about tests? ;-)

**Comment by tadatuta** [comment]
Tests and documentation added, ready for merge

**Comment by zxqfox** [comment]
u r z bist!

File: index.js [hunk 1]
```diff
@@ -76,42 +91,61 @@ BemConfig.prototype.get = function() {
...
+        let conf = allConfigs[i];
+        let levels = conf.levels || {};
 
-            Object.assign(levelOpts, conf);
+        Object.assign(levelOpts, conf);
 
-            for (let level in levels) {
-                if (level === absLevelPath) {
+        for (let level in levels) {
+            if (level === absLevelPath) {
```
Comment (zxqfox) [inline index.js]: `if (level !== absLevelPath) continue;` ?

File: index.js [hunk 2]
```diff
@@ -76,42 +91,61 @@ BemConfig.prototype.get = function() {
...
-                    levelOpts = _.mergeWith({}, levels[level], levelOpts, (objValue, srcValue) => {
-                        if (Array.isArray(objValue)) {
-                            return srcValue;
-                        }
-                    });
-                }
+                // works like deep extend but overrides arrays
+                levelOpts = _.mergeWith({}, levels[level], levelOpts, (objValue, srcValue) => {
+                    if (Array.isArray(objValue)) {
+                        return srcValue;
```
Comment (zxqfox) [inline index.js]: `(objValue, srcValue) => Array.isArray(objValue) && srcValue;`

File: index.js [hunk 3]
```diff
@@ -76,42 +91,61 @@ BemConfig.prototype.get = function() {
...
+                });
             }
-
-            if (conf.root) { break; }
         }
 
-        delete levelOpts.__source;
-        delete levelOpts.levels;
-        delete levelOpts.root;
+        if (conf.root) { break; }
```
Comment (zxqfox) [inline index.js]: There should be an exception about break/continue/return statements in ifs.

File: index.js [hunk 4]
```diff
@@ -125,22 +159,50 @@ BemConfig.prototype.library = function(libName) {
...
+/**
  * Returns map of settings for each of level
  * @returns {Promise}
  */
 BemConfig.prototype.levelMap = function() {
     return this.get().then(config => {
         const projectLevels = config.levels;
 
-        return Promise.all(Object.keys(config.libs).map(libName => {
+        return Promise.all((config.libs ? Object.keys(config.libs) : []).map(libName => {
```
Comment (zxqfox) [inline index.js]: It's a bugfix about another things?

Comment (tadatuta) [inline index.js]: true. you're about moving it to another PR?

Comment (zxqfox) [inline index.js]: Not really, we are humans. We are not ideal. But we can write the ideal code. We are just lazy also, so nvm.

Comment (zxqfox) [inline index.js]: Oh, but I'd prefer to move it to separate commit. You are lazier than me, lel.

**Comment by zxqfox** [comment]
Actually, I'm fine with this version and would merge it.

@yeti-or, @blond Fellows, take a look, please!

File: index.js (hunk 2 above)
Comment (tadatuta) [inline index.js]: won't work as `_.mergeWith({ a: 1, b: 2 }, false)` results with `{ a: false, b: false }`

Comment (zxqfox) [inline index.js]: Maybe undefined?

**Comment by tadatuta** [comment]
Split into commits, fixed @zxqfox notice

File: index.js (hunk 2 above)
Comment (tadatuta) [inline index.js]: `(objValue, srcValue) => Array.isArray(objValue) && srcValue || undefined`?
isn't it ugly?

Comment (zxqfox) [inline index.js]: No, if it works.

**Comment by voischev** [comment]
👍
